integration-tests/delegatable-credentials.test.ts

```diff
@@ -0,0 +1,282 @@
+const SUBJECT_DID = 'did:example:alice';
+
+const CHALLENGE = 'test-challenge-123';
+const DOMAIN = 'test.example.com';
+
+describe('Delegatable Credentials', () => {
+  let rootIssuerKey: any;
+  let delegateKey: any;
+  let rootIssuerDid: string;
+  let delegateDid: string;
+  let delegationCredential: any;
+  let creditScoreCredential: any;
+  let unauthorizedDelegationCredential: any;
+
+  beforeAll(async () => {
+    // Generate key pairs for root issuer and delegate
+    rootIssuerKey = await didService.generateKeyDoc({
+      type: 'ed25519',
+    });
+    delegateKey = await didService.generateKeyDoc({
+      type: 'ed25519',
+    });
+
+    // Extract DIDs from the key documents
+    rootIssuerDid = rootIssuerKey.controller;
+    delegateDid = delegateKey.controller;
+
+    console.log('Root Issuer DID:', rootIssuerDid);
+    console.log('Delegate DID:', delegateDid);
+
+    // Issue the root delegation credential
+    // This grants the delegate authority to issue creditScore claims
+    delegationCredential = await issueDelegationCredential(rootIssuerKey, {
+      id: DELEGATION_ROOT_ID,
+      issuerDid: rootIssuerDid,
+      delegateDid: delegateDid,
+      mayClaim: ['creditScore'],
+      context: CREDIT_SCORE_DELEGATION_CONTEXT,
+      types: ['VerifiableCredential', 'CreditScoreDelegation', 'DelegationCredential'],
+      additionalSubjectProperties: {
+        body: 'Issuer of Credit Scores',
+      },
+    });
+
+    // Issue a credit score credential as the delegate
+    creditScoreCredential = await issueDelegatedCredential(delegateKey, {
+      id: CREDIT_SCORE_CRED_ID,
+      issuerDid: delegateDid,
+      subjectDid: SUBJECT_DID,
+      claims: {
+        creditScore: 760,
+      },
+      rootCredentialId: DELEGATION_ROOT_ID,
+      previousCredentialId: DELEGATION_ROOT_ID,
+      context: CREDIT_SCORE_CREDENTIAL_CONTEXT,
+      types: ['VerifiableCredential', 'CreditScoreCredential'],
+    });
+
+    // Issue an unauthorized delegation (no creditScore in mayClaim)
+    unauthorizedDelegationCredential = await issueDelegationCredential(rootIssuerKey, {
+      id: 'urn:cred:unauthorized-delegation',
+      issuerDid: rootIssuerDid,
+      delegateDid: delegateDid,
+      mayClaim: ['someOtherClaim'], // Does NOT include creditScore
+      context: CREDIT_SCORE_DELEGATION_CONTEXT,
+      types: ['VerifiableCredential', 'CreditScoreDelegation', 'DelegationCredential'],
+    });
+  });
+
+  it('should issue a valid delegation credential', () => {
+    expect(delegationCredential).toBeDefined();
+    expect(delegationCredential.id).toBe(DELEGATION_ROOT_ID);
+    expect(delegationCredential.issuer).toBe(rootIssuerDid);
+    expect(delegationCredential.credentialSubject.id).toBe(delegateDid);
+    expect(delegationCredential.credentialSubject[MAY_CLAIM_IRI]).toContain('creditScore');
+    expect(delegationCredential.proof).toBeDefined();
+    expect(delegationCredential.rootCredentialId).toBe(DELEGATION_ROOT_ID);
```

Comment:
if it's the top level credential, are we expecting it to reference itself as the rootCredentialId? I think I was expecting it to be empty/null for the root. But, I must admit I'm not sure what Sam is doing on the API side for this

Reply:
I'm defining that logic in the test; the SDK does not enforce it being required. So it's up to us actually

Reply:
It works this way: https://github.com/docknetwork/wallet-sdk/pull/445/commits/20ae2f12c7b2594a270422c8eca21088536d8c36